fabric-convention-tags-v2/src/client/java/net/fabricmc/fabric/impl/tag/convention/ConventionLogWarningsClient.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.convention;
+
+import java.util.List;
+import java.util.Locale;
+
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.client.resource.language.I18n;
+import net.minecraft.item.Item;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.fabricmc.loader.api.FabricLoader;
+
+public class ConventionLogWarningsClient implements ClientModInitializer {
```

Comment:
I think this should be in the server as well? We do load mod-provided locales there. At least if it's in client, make it log only once.

Reply:
I18n is a clientsided only class. How would I check if a translation exist on a dedicated server?

The logging only happens when once when a world is loaded up. And since this is more for debugging and opt-in, I don't think we need to store an extra field somewhere to make it only run once ever.

Reply:
[3fca778](https://github.com/FabricMC/fabric/pull/3716/commits/3fca778f629810d6b04f29991cc75159dc842f00)